W-22402743-rtfctl-3.0.131-release-kt (#1697)
mulesoft/docs-runtime-fabric: modules/ROOT/pages/manage-agent-truststore-rtfctl.adoc

= Managing the Agent Truststore with RTFCTL

> The `rtfctl agent-truststore` command group lets you inspect and update the certificate authority (CA) certificates that the Runtime Fabric agent uses to establish mutually authenticated TLS (mTLS) connections with the Anypoint Platform control plane.

Reviewer: ```suggestion
Use the `rtfctl agent-truststore` command group to inspect and update the certificate authority (CA) certificates that the Runtime Fabric agent uses to establish mutually authenticated TLS (mTLS) connections with the Anypoint Platform control plane.
```

The `rtfctl agent-truststore` commands provide these operations directly through `rtfctl`, without requiring any additional tooling.

== Commands

include::partial$rtfctl-commands.adoc[tag=agent-truststore-rtfctl,leveloffset=+1]

== Workflows

=== Inspect the Current Truststore

Before making changes, review which certificates are loaded and when they expire:

----
rtfctl agent-truststore info --namespace rtf
----

Use `-o json` to get machine-readable output suitable for scripting or CI checks.

=== Add a New CA to the Truststore (Interactive)

Use `refresh-secret-with-pem` to retrieve the current truststore from the cluster, merge a new PEM file into both the PKCS#12 and PEM stores, preview the result, and upload it back in a single step:

----
rtfctl agent-truststore refresh-secret-with-pem -c new-ca.pem --namespace rtf
----

> The command prints the subjects of all certificates that will be uploaded and asks for confirmation before patching the secret.

Reviewer: ```suggestion
The command prints the subjects of all certificates to upload and asks for confirmation before patching the secret.
```
